src/components/shared/MobileMenu/MenuItem.tsx

```diff
@@ -0,0 +1,181 @@
+
+type LinkItem = Partial<Record<'href' | 'to', string>>
+
+export interface MenuItemProps extends LinkItem {
+  title: string
+  hint?: string
+  children?: MenuItemProps[]
+  component?: React.ElementType
+  divider?: boolean
+  menuProps?: BoxProps
+  onClick?: React.MouseEventHandler
+  Menu?: MenuType
+  disable?: boolean
+  notificationInfo?: NotificationBadgeType
+  showNotificationCount?: boolean
+}
+
+export const MenuItem = ({
+  title,
+  hint,
+  disable,
+  children,
+  divider,
+  component = Link,
+  menuProps,
+  onClick,
+  Menu,
+  notificationInfo,
+  showNotificationCount,
+  ...props
+}: MenuItemProps) => {
+  const { spacing } = useTheme()
+  const visible = useSelector(appearMenuSelector)
+  const dispatch = useDispatch()
+  const [open, setOpen] = useState(false)
+  const notificationColor = notificationInfo?.isNotificationAlert
+    ? '#FB6540'
+    : '#B3CB2D'
+
+  return (
+    <ListItem
+      sx={{
+        display: 'block',
+        position: 'relative',
+        padding: spacing(0, 1),
+        ...(onClick != null && { cursor: 'pointer' }),
+      }}
+      onClick={() => {
+        if (children != null) {
+          setOpen(!open)
+        } else {
+          dispatch(setAppear({ MENU: !visible }))
+        }
+      }}
+    >
+      <Link
+        component={children == null ? component : Box}
+        sx={{
+          color: `${disable ? 'text.disabled' : 'text.primary'}`,
+          pointerEvents: `${disable ? 'none' : 'auto'}`,
+          display: 'flex',
+          justifyContent: 'space-between',
+          alignItems: 'center',
+          padding: spacing(hint ? 1.3 : 1.5, 2),
+          borderRadius: 3,
+          typography: 'label3',
+          textDecoration: 'none',
+          whiteSpace: 'nowrap',
+          ':hover': {
+            backgroundColor: 'selected.hover',
+            color: 'primary.dark',
+            '.MuiSvgIcon-root': {
+              color: 'primary.dark',
+            },
+          },
+        }}
+        {...props}
+      >
+        {title}
+        {children != null && !open && (
+          <KeyboardArrowRightIcon sx={{ color: 'icon.icon02' }} />
+        )}
+        {children != null && open && (
+          <KeyboardArrowDownIcon sx={{ color: 'icon.icon02' }} />
+        )}
+        {showNotificationCount &&
+          notificationInfo != null &&
+          notificationInfo.notificationCount > 0 && (
+            <Typography
+              sx={{
+                fontWeight: '500',
+                fontSize: '0.75rem',
+                minWidth: '20px',
+                padding: '2px 6px',
+                height: '20px',
+                borderRadius: '10px',
+                background: notificationColor,
+                color: 'white',
+              }}
+            >
+              {notificationInfo?.notificationCount}
+            </Typography>
+          )}
+      </Link>
+      {Menu != null && children != null && open && (
```

Comment:
I think we don't need to check if children are null or undefined here because two lines below we have children?